### ISSUE TITLE
Adding Custom Attributors to Docs

### ISSUE THREAD

**Issue by DennisSmolek** [quilljs/quill#1832]
I'm using Quill.js for a photoshop style online editor as my text tool and I've been struggling with adding my own style attributors.

The docs show how to import existing ones but I couldn't figure out how to create my own.

I stumbled onto: #1036 by @benbro but he and @jhchen decided it wasn't needed in the docs because of no real-world example for custom classes.

I wanted to add my case for simply adding the style `line-height`. Iy took me a while as first I was importing Parchment into my project just to create the attributor and the was confused because I know Quill already had somewhere, so then I had to go through the source code to figure it all out.

The Typescript definitions don't include the `Quill.imports` like mentioned [HERE](https://quilljs.com/guides/how-to-customize-quill/#extending-blots) So it took me a lot of trial and error. 

I think adding a section on creating your own classes or style additions would be very helpful. 

Something like: 

### Adding your own Inline styles
If you need to customize blah blah blah....

``` javascript
const Parchment = Quill.import('parchment');
const LineStyle = new Parchment.Attributor.Style('lineHeight', 'line-height', {
      scope: Parchment.Scope.INLINE,
             whiteList: ['10px', '12px', '16px']
      });
Quill.register(LineStyle, true);
```

**Comment by jhchen** [comment]
Did you stumble across https://github.com/quilljs/parchment#attributors? It lists not just one example but every example for each type of attributor you can add so not sure what additions your example includes.

**Comment by DennisSmolek** [comment]
Yes, that's how I learned how to do it within parchment. The point I was making is that I think at least style or class Attributors should be discussed within the **Quill** documentation.

I think adding css beyond `color` & `size` is a common need and the documentation does a good job explaining how to add fonts to the whitelist but it does a poor job explaining how to add your own classes or styles which (I feel) are very useful and likely what folks are trying to do.

I think instead of just sending folks to Parchment in the very last paragraph there should be a basic example and then "Get even more details at the Parchment Readme" if you're trying to keep the docs short.

Right now if I wanted to learn how to add my own class `.awesome` when I click a button Id:

1.  [Go to the how to customize quill page](https://quilljs.com/guides/how-to-customize-quill/)
2. Read about styles and classes.
3. Read about Blots (which I don't need)
4. Read that if I want to do my Attributor own I need to...
5. Go to Parchment Readme (at the very top) get confused so go back and...
6. Go to the cloning Medium article, which shows how to add to existing attributors but not custom ones
7. Go back to Parchment's docs, learn everything about how all of the code works and finally at the last section learn about the class/style attributors
8. Try importing them with `Quill` but get rejected because the Typescript Definition doesn't have `.imports` [like the docs say](https://quilljs.com/guides/how-to-customize-quill/#extending-blots)
9. Manually import all of parchment on my own.
10. Feel like I'm already importing Parchment when I import Quill so tear through the source code to figure out what I'm doing wrong.
11. Get frustrated and create Github Issue...

Just looking to help out others and save them some time...

**Comment by jhchen** [comment]
The major challenge is that the documentation is it applied to a broad audience. You happened to pick the worst case of your own contrived example. If I optimized the docs for one person I could easily make that one step. In this case if that person was you, I would remove the info about styles and classes, blots, etc. That of course is not globally optimal since there are many people that actually care about vanilla CSS customization or thought they wanted something more complicated like attributors, but after reading that section conclude the simpler approach is better. 

I have a hard time taking your complaint seriously when you sprinkle in falsehoods like number 8. Nowhere in the associated link does Quill talk about Typescript; Quill is written in Javascript and does not claim to supply Typescript type definitions. If you are talking about DefinitelyTyped, that project has no affiliation with Quill and contributions are community driven.

**Comment by DennisSmolek** [comment]
@jhchen Hey Jason, I'm sorry if I came off being harsh or judgemental, that wasn't my intention at all and I'm sorry if I came off that way. I love Quill and it's instrumental in my application and I appreciate all the hard work put into it.

I admit, I did struggle and was frustrated when I first wrote the issue, but I did not mean to seem confrontational. 
Adding custom attributors is very powerful, and is baked into Quill through parchment, but wasn't easy to find. If you don't think it important, no worries, it's your project.

As to falseness, that is my fault for being confusing. What I was saying is that in [that link](https://quilljs.com/guides/how-to-customize-quill/#extending-blots) Quill documents using `Quill.imports`

But as I'm using typescript if I try to use `Quill.imports` it throws an error saying `imports is not a method of Quill` so I assumed that it did not work.

I wasn't saying anything about Quill/Typescript I was talking about `Quill.imports`
